engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/node/ForeignEvalNode.java

```diff
@@ -14,15 +14,15 @@
 import org.enso.interpreter.epb.EpbContext;
 import org.enso.interpreter.epb.EpbLanguage;
 import org.enso.interpreter.epb.EpbParser;
+import org.enso.interpreter.epb.runtime.ForeignParsingException;
 import org.enso.interpreter.epb.runtime.GuardedTruffleContext;
+import org.graalvm.polyglot.Context;
 
 public class ForeignEvalNode extends RootNode {
   private final EpbParser.Result code;
   private @Child ForeignFunctionCallNode foreign;
   private @Child ContextRewrapNode rewrapNode = ContextRewrapNode.build();
-  private @Child ContextRewrapExceptionNode rewrapExceptionNode =
-      ContextRewrapExceptionNode.build();
-  private @CompilerDirectives.CompilationFinal AbstractTruffleException parseError;
+  private @CompilationFinal ForeignParsingException parseException;
```

Comment:
Is there any reason to throw the exception on _fast path_? Just `transferToInterpreter()` and throw it _"slowly"_.